### DISCUSSION TITLE
How to handle the creation of a new IDS resource?

### DISCUSSION THREAD

**Discussion by MikhailGordienk** [eclipse-edc/Connector#108] [Q&A]
Hello!

There is no API to create a resource in the _ids-api-catalog_ module. Does that mean resource creation must be handled by extensions?

I think it could be because in some cases there might be a lot of steps to do to get data indexed. For instance, the one
might want to share the entire S3 bucket with many files and folders inside, it would take some time to process them all and publish them as catalog entries. And this "indexing" would be platform-specific.

Do you plan to have any related interfaces/abstractions in the _core_?

**Comment by jimmarino** [comment]
AssetIndex will be replacing MetaDataStore. An AssetIndex is a facade over some form of asset storage, which is generally an external system. The external system may implement its own schema, query mechanism, persistence, and loading mechanisms. Given this, indexing and storage of assets are out-of-scope from the perspective of the core.

Another design principle is that the external system backing the AssetIndex will not likely store the actual asset contents. Instead, it will keep a dereferenceable pointer to the asset contents. This makes it possible to use a database to store the index while data content can be stored in S3, Azure storage, a file system, etc. It also improves security since access to the data is segmented from access to the index.

**Comment by DominikPinsel** [comment]
Hello everyone,

I just wanted to mention that this discussion is probably related to the issue #118, where I suggested to remove the possibility to create new resources from within the connector (for now). If there are reasons not to do this, please comment :)

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sub>
